### ISSUE TITLE
JDBC Datasource property should allow to select or create a bean

### ISSUE THREAD

**Issue by lhein** [KaotoIO/kaoto-next#967]

**Comment by lordrip** [comment]
Ok, if I'm understanding correctly, in this case, the `dataSource` field should allow to create or select existing beans but without having a `#` prefix.

**Comment by lhein** [comment]
It can be also a predefined constant...see above description

**Comment by lordrip** [comment]
yup, I saw that part as well. Judging by the list of custom features this field might require, I think it would be better to provide something custom for this case. It could also automatically add predefined beans for known databases.

**Comment by mmelko** [comment]
Datasource beans are often configured outside of the camel in the spring-boot and quarkus configurations.  I'm not sure if it's the best practice to add datasource configuration into the camel file.

- If configured in the `beans` it shouldn't provide list of all defined beans only datasource types.  
-  new Bean form should have predefined datasource type, and also properties like name, serverURL, password  and driver(?)

so for the "quick fix" just having set the `default/dataSource` value  to tell the camel to look for the DS or leaving custom input field. ?
